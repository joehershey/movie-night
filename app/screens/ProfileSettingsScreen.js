import React from "react";
import {
  StyleSheet,
  View,
  Text,
  TouchableWithoutFeedback,
  SafeAreaView,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";

import TopBar from "../components/TopBar";
import ChangePassword from "../components/ChangePassword";

function ProfileSettingsScreen(props) {
  const [showPopup, toggleShowPopup] = React.useState(false);

  return (
    <SafeAreaView style={STYLES.container}>
      {/* Top Bar */}
      <TopBar
        navigation={props.navigation}
        screen="Landing"
        message="Groups"
      ></TopBar>
      <ChangePassword
        showPopup={showPopup}
        token={props.token}
        toggleShowPopup={toggleShowPopup}
        url={props.url}
        user_id={props.user_id}
      />

      {/* Content */}
      <View style={styles.content}>
        <Text style={{ fontSize: 50, color: "white" }}>
          {"Profile Settings"}
        </Text>
        <Text style={{ fontSize: 30, color: "white" }}>
          {"Username: " + props.username}
        </Text>

        <View style={{ width: "100%", alignItems: "center" }}>
          <TouchableWithoutFeedback
            testID="ChangePasswordButton"
            onPress={() => toggleShowPopup(!showPopup)}
          >
            <View
              style={[
                STYLES.settingsButtons,
                STYLES.btn,
                { backgroundColor: COLORS.secondary },
              ]}
            >
              <Text style={[{ color: "white", fontSize: 20 }]}>
                Change Password
              </Text>
            </View>
          </TouchableWithoutFeedback>

          <TouchableWithoutFeedback
            testID="NavigateButton"
            onPress={() => props.navigation.navigate("Welcome")}
          >
            <View
              style={[
                STYLES.settingsButtons,
                STYLES.btn,
                { backgroundColor: "white" },
              ]}
            >
              <Text style={[{ color: COLORS.primary, fontSize: 20 }]}>
                Log out
              </Text>
            </View>
          </TouchableWithoutFeedback>
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  content: {
    flex: 12,
    flexDirection: "column",
    justifyContent: "space-evenly",
    alignItems: "center",
  },
});

export default ProfileSettingsScreen;
