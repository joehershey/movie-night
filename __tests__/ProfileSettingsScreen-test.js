import ProfileSettingsScreen from '../app/screens/ProfileSettingsScreen';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test Profile Settings Screen', () => {
    beforeEach( async () => {
        global.tree = renderer.create(<ProfileSettingsScreen />).toJSON();
    });

    it("Profile settings screen renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });
});