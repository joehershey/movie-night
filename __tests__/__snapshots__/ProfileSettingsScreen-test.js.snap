// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile Settings Screen Profile settings screen renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "margin": 5,
          }
        }
      >
        Groups
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "paddingBottom": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginBottom": 0,
            "width": 100,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View>
    <View>
      <Modal
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.5)",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 22,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderColor": "#ccc",
                "borderRadius": 20,
                "elevation": 5,
                "height": "50%",
                "margin": 20,
                "padding": 25,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 30,
                  "paddingBottom": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                Change Password
              </Text>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              Current Password:
            </Text>
            <View
              style={
                Object {
                  "height": 45,
                  "marginBottom": 5,
                  "width": 275,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Current Password"
                placeholderTextColor="#D3D3D3"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "auto",
                    "justifyContent": "center",
                    "margin": 4,
                    "padding": 10,
                    "width": "auto",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              New Password:
            </Text>
            <View
              style={
                Object {
                  "height": 45,
                  "marginBottom": 5,
                  "width": 275,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="New Password"
                placeholderTextColor="#D3D3D3"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "auto",
                    "justifyContent": "center",
                    "margin": 4,
                    "padding": 10,
                    "width": "auto",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              Confirm New Password:
            </Text>
            <View
              style={
                Object {
                  "height": 45,
                  "marginBottom": 5,
                  "width": 275,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="New Password"
                placeholderTextColor="#D3D3D3"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": "auto",
                    "justifyContent": "center",
                    "margin": 4,
                    "padding": 10,
                    "width": "auto",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#166fdb",
                      "height": 60,
                      "top": 10,
                      "width": "40%",
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "margin": 5,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignContent": "center",
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Submit
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1d1f1e",
                      "height": 60,
                      "top": 10,
                      "width": "40%",
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "margin": 5,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignContent": "center",
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Close
                </Text>
              </View>
            </View>
          </View>
        </View>
      </Modal>
      <Modal
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.5)",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 22,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderColor": "#ccc",
                "borderRadius": 20,
                "elevation": 5,
                "height": "50%",
                "margin": 20,
                "padding": 25,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "paddingBottom": 30,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                You have changed your password!
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#1d1f1e",
                    "height": 60,
                    "top": 10,
                    "width": "40%",
                  },
                  Object {
                    "alignItems": "center",
                    "borderRadius": 20,
                    "justifyContent": "center",
                    "margin": 5,
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "alignContent": "center",
                    "color": "white",
                    "fontSize": 20,
                  }
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <View>
    <View>
      <Modal
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.5)",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 22,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderColor": "#ccc",
                "borderRadius": 20,
                "elevation": 5,
                "height": "50%",
                "margin": 20,
                "padding": 25,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "paddingBottom": 30,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                Are you sure you would like to delete your account?
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#166fdb",
                      "height": 60,
                      "top": 10,
                      "width": "40%",
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "margin": 5,
                    },
                    Object {
                      "backgroundColor": "#b81f00",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignContent": "center",
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Confirm
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1d1f1e",
                      "height": 60,
                      "top": 10,
                      "width": "40%",
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "margin": 5,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignContent": "center",
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Close
                </Text>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 12,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 50,
        }
      }
    >
      Profile Settings
    </Text>
    <Text
      style={
        Object {
          "fontSize": 30,
        }
      }
    >
      Username: undefined
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#166fdb",
              "height": 70,
              "top": 10,
              "width": "60%",
            },
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "justifyContent": "center",
              "margin": 5,
            },
            Object {
              "backgroundColor": "#1d1f1e",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Change Password
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#166fdb",
              "height": 70,
              "top": 10,
              "width": "60%",
            },
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "justifyContent": "center",
              "margin": 5,
            },
            Object {
              "backgroundColor": "#1d1f1e",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Log out
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#166fdb",
              "height": 70,
              "top": 10,
              "width": "60%",
            },
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "justifyContent": "center",
              "margin": 5,
            },
            Object {
              "backgroundColor": "#b81f00",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Delete Account
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
